internal/conversion/frr_conversion.go

```diff
@@ -148,18 +152,39 @@ func neighborToFRR(n v1alpha1.Neighbor) (*frr.NeighborConfig, error) {
 	}
 	res.HoldTime, res.KeepaliveTime, err = parseTimers(n.HoldTime, n.KeepaliveTime)
 	if err != nil {
-		return nil, fmt.Errorf("invalid timers for neighbor %s, err: %w", neighborName(n), err)
+		return nil, nil, fmt.Errorf("invalid timers for neighbor %s, err: %w", neighborName(n), err)
 	}
 
 	if n.ConnectTime != nil {
 		connectSecond, err := durationToUint64(n.ConnectTime.Duration / time.Second)
 		if err != nil {
-			return nil, fmt.Errorf("invalid connecttime %v: %w", n.ConnectTime.Duration, err)
+			return nil, nil, fmt.Errorf("invalid connecttime %v: %w", n.ConnectTime.Duration, err)
 		}
 		res.ConnectTime = ptr.To(connectSecond)
 	}
 
-	return res, nil
+	if n.BFD == nil {
+		return res, nil, nil
+	}
+	res.BFDEnabled = true
+	if ptr.AllPtrFieldsNil(n.BFD) {
+		return res, nil, nil
+	}
+
+	profileName := fmt.Sprintf("neighbor-%s", n.Address)
+	res.BFDProfile = profileName
+	bfdProfile := &frr.BFDProfile{
+		Name:             profileName,
+		ReceiveInterval:  n.BFD.ReceiveInterval,
+		TransmitInterval: n.BFD.TransmitInterval,
+		DetectMultiplier: n.BFD.DetectMultiplier,
+		EchoInterval:     n.BFD.EchoInterval,
+		EchoMode:         ptr.Deref(n.BFD.EchoMode, false),
+		PassiveMode:      ptr.Deref(n.BFD.PassiveMode, false),
+		MinimumTTL:       n.BFD.MinimumTTL,
+	}
```

Comment:
this is a bit odd (returning the profile as a side-effect as this function), can we modify the `NeighborConfig` struct to have the type of its BFDProfile field as `BFDProfile` instead of `string`, and populate it here instead (with the cost of modifying the frr tmpls)?
alternatively, we can just move this to a different function 😅

Reply:
that was my original version (embedding it), but then I realized the profile is not related to the neighbor, it's a global thing in frr.
I will move to a separate function.